test/test_find_library.cpp

```diff
@@ -59,9 +59,18 @@ TEST(test_find_library, find_library)
 #endif
 
   // Positive test.
-  const std::string test_lib_actual = find_library_path("test_library");
+  std::string test_lib_actual = find_library_path("test_library");
   EXPECT_EQ(test_lib_actual, expected_library_path);
 
+#if !defined(_WIN32) && !defined(__APPLE__)
+  EXPECT_EQ(setenv(env_var, "", override), 0);
+
+  // Test with empty LD_LIBRARY_PATH to emulate setcap / setuid executable.
+  // Using RUNPATH to find library instead.
+  test_lib_actual = find_library_path("test_library");
+  EXPECT_EQ(test_lib_actual, expected_library_path);
+#endif
+
```

Comment:
Could we move this into a different test case, and use the gtest fixture to clear out and re-enable the `LD_LIBRARY_PATH` value? This does not re-enable the environment variable, so all tests after this one will actually be running in a modified environment. It will also potentially break people relying on gtest flags such as `--gtest_repeat=10` (first one will have full env, next ones will run w/o `LD_LIBRARY_PATH`).

Reply:
How about moving the setting of the env var to the test properties in CMakeLists and reusing the test function for both cases? This would also be more in line with what's actually happening (`LD_LIBRARY_PATH` being dropped at execution). Like this:

```cmake
  if(WIN32)
    SET(LIBRARY_PATH_VAR "PATH")
  elseif(APPLE)
    SET(LIBRARY_PATH_VAR "DYLD_LIBRARY_PATH")
  else()
    SET(LIBRARY_PATH_VAR "LD_LIBRARY_PATH")
  endif()

  add_library(test_library SHARED test/test_library.cpp)

  ament_add_gtest(test_find_library test/test_find_library.cpp)
  target_link_libraries(test_find_library ${PROJECT_NAME} test_library)
  set_tests_properties(test_find_library PROPERTIES
    ENVIRONMENT
      "${LIBRARY_PATH_VAR}=$<TARGET_FILE_DIR:test_library>;_TEST_LIBRARY=$<TARGET_FILE:test_library>")

  if(NOT (WIN32 OR APPLE))
    ament_add_gtest(test_find_library_runpath test/test_find_library.cpp)
    target_link_libraries(test_find_library_runpath ${PROJECT_NAME} test_library)
    set_tests_properties(test_find_library_runpath PROPERTIES
      ENVIRONMENT
        "${LIBRARY_PATH_VAR}=;_TEST_LIBRARY=$<TARGET_FILE:test_library>")
  endif()
```

Reply:
This will still have a global effect and impact other test cases, so this does not really solve the problem at hand IMHO.

Reply:
Hmm, on https://cmake.org/cmake/help/latest/prop_test/ENVIRONMENT.html it says
> The environment is restored to its previous state after the test is done.

The env setting code in `test_find_library.cpp` would be removed as this would be handled through the test property in CMake that sets (and restores) the `LIBRARY_PATH_VAR` directly.

Reply:
Ok, I pushed this change for review and further discussion.